src/components/UploadCSV/UploadCSV.jsx

```diff
@@ -0,0 +1,159 @@
+
+  useEffect(() => {
+    if (isUploadingNewFile) {
+      setUploadErrors([]);
+      setIsLoading(true);
+    }
+  }, [isUploadingNewFile]);
+
+  const checkEmptyCells = async (
+    line,
+    dateCSV,
+    boxNumberCSV,
+    zipCodeCSV,
+    launchedOrganicallyCSV,
+  ) => {
+    const emptyCells = [];
+    if (!dateCSV) {
+      emptyCells.push('date');
+    }
+    if (!boxNumberCSV) {
+      emptyCells.push('box number');
+    } else {
+      const res = await FYABackend.get(`/boxForm/exists/${boxNumberCSV}`);
+      console.log(res.data);
+      if (res.data) {
+        console.log('BOX NUMBER EXISTS');
+        setUploadErrors(prevState => [...prevState, `box number ${boxNumberCSV} already exists`]);
+      }
+    }
+    if (!zipCodeCSV) {
+      emptyCells.push('zip code');
+    }
+    if (!launchedOrganicallyCSV) {
+      emptyCells.push('launched organically');
+    }
+    emptyCells.map(cell =>
+      setUploadErrors(prevState => [...prevState, `missing ${cell} in line ${line}`]),
+    );
+    // if (emptyCells.length > 0) setIsLoading(false);
+    setIsLoading(false);
+  };
+
+  const readCSV = () => {
+    readRemoteFile(CSVFile, {
+      complete: results => {
+        // parse each line in csv file and upload each to the backend
+        console.log(results.data);
+        for (let i = 1; i < results.data.length; i += 1) {
+          const dateCSV = results.data[i][0];
+          const boxNumberCSV = results.data[i][1];
+          const zipCodeCSV = results.data[i][2];
+          const launchedOrganicallyCSV = results.data[i][3];
+          // TODO: check if any cells are empty
+          checkEmptyCells(i, dateCSV, boxNumberCSV, zipCodeCSV, launchedOrganicallyCSV);
+          // checkErrors(results.data[i], i);
+          // TODO: validate zipCodeCSV (wait until common/utils is updated)
+          // TODO: validate date?
+          // TODO: validate box number is type integer
+          setFormDatas(prevState => [
+            ...prevState,
+            {
+              boxNumber: boxNumberCSV,
+              date: dateCSV,
+              zipCode: zipCodeCSV,
+              boxLocation: '',
+              message: '',
+              picture: '',
+              comments: '',
+              launchedOrganically: launchedOrganicallyCSV.toLowerCase() === 'yes',
+            },
+          ]);
+        }
+
+        setIsUploadingNewFile(false);
+        setCSVFile();
+        // setFormData();
+      },
+    });
+  };
+
+  const onUpload = e => {
+    e.preventDefault();
+    if (CSVFile) {
+      setCSVFilename(CSVFile.name);
+      readCSV();
+    }
+  };
+
+  const addToMap = async e => {
+    e.preventDefault();
+    console.log('ADD TO MAP');
+    return Promise.all(
+      formDatas.map(async formData => {
+        console.log('FORMDATA: ', formData);
+        await FYABackend.post('/boxForm', formData, {
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        });
+      }),
+    );
```

Comment:
Using `Promise.all` here is smart, but considering that they will be uploading thousands of boxes at a time, a separate backend route should be created to handle mass uploads. That way the all of the form data can be sent at once to the backend, which can process it as a single request.

Reply:
I would merge this branch in now, and resolve this in a separate PR (frontend and backend)

Reply:
Good idea! I just added a route in the backend to handle this and fixed the frontend